cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java

```diff
@@ -121,6 +129,8 @@ public void fetch(HttpRequest request, final HttpResponder responder,
       return;
     }
 
+    // Make sure the user has READ permission on the stream since getConfig doesn't check for the same.
+    authorizationEnforcer.enforce(streamId.toEntityId(), authenticationContext.getPrincipal(), Action.READ);
```

Comment:
why does this logic have to be in the handler? Can it not be at a layer that the handler uses, like for other entities?